src/core/LanguageMenu/LanguageMenu/LanguageMenu.tsx

```diff
@@ -0,0 +1,236 @@
+import React, { Component, ReactNode, Fragment } from 'react';
+import { default as styled } from 'styled-components';
+import classnames from 'classnames';
+import {
+  Menu,
+  MenuButton,
+  MenuItems,
+  MenuPopover,
+  MenuPopoverProps,
+} from '@reach/menu-button';
+import { positionDefault } from '@reach/popover';
+import { PRect } from '@reach/rect';
+import { classnamesValue } from '../../../utils/typescript';
+import { logger } from '../../../utils/logger';
+import { HtmlSpan } from '../../../reset/HtmlSpan/HtmlSpan';
+import { Icon } from '../../Icon/Icon';
+import {
+  baseStyles,
+  languageMenuPopoverStyles,
+} from './LanguageMenu.baseStyles';
+
+const baseClassName = 'fi-language-menu';
+const languageMenuClassNames = {
+  item: `${baseClassName}_item`,
+  itemLang: `${baseClassName}-language_item`,
+  button: `${baseClassName}_button`,
+  buttonOpen: `${baseClassName}-language_button_open`,
+  buttonLang: `${baseClassName}-language_button`,
+  popover: `${baseClassName}_popover`,
+  popoverLang: `${baseClassName}-language_popover`,
+  iconLang: `${baseClassName}-language_icon`,
+};
+
+export interface LanguageMenuItemBaseProps {
+  /** Operation to run on select */
+  onSelect: () => void;
+  /** Item content */
+  children: ReactNode;
+}
+
+type SupportedMenuLinkComponent = keyof JSX.IntrinsicElements;
+
+interface LanguageMenuLinkPropsWithType {
+  type: 'menulink';
+  /** Url to direct to */
+  href: string;
+  /** Item content */
+  children: ReactNode;
+  className?: string;
+  as?: SupportedMenuLinkComponent;
+}
+
+export interface LanguageMenuLinkBaseProps
+  extends Omit<LanguageMenuLinkPropsWithType, 'type'> {}
+
+export type LanguageMenuPopoverItemsProps =
+  | LanguageMenuItemBaseProps
+  | LanguageMenuLinkPropsWithType;
+type OptionalLanguageMenuPopoverProps = {
```

Comment:
This could probably be written as Partial<MenuPopoverProps>, but should not make any difference if left as is.